analyzers/src/SonarAnalyzer.CSharp/Rules/CloudNative/AzureFunctionsCatchExceptions.cs

```diff
@@ -0,0 +1,100 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Helpers;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    public static class Fixme   // FIXME: Remove temporary workaround
+    {
+        public static bool IsAzureFunction(this SyntaxNodeAnalysisContext context) =>
+            context.ContainingSymbol.GetAttributes().Any(x => x.AttributeClass.Name.Contains("FunctionName"));
+    }
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class AzureFunctionsCatchExceptions : SonarDiagnosticAnalyzer
+    {
+        private const string DiagnosticId = "S6421";
+        private const string MessageFormat = "Wrap Azure Function body in try/catch block.";
+
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+                {
+                    if (c.IsAzureFunction())
+                    {
+                        var method = (MethodDeclarationSyntax)c.Node;
+                        var walker = new Walker();
+                        if (walker.SafeVisit(c.Node) && walker.HasInvocationOutsideTryCatch)
+                        {
+                            c.ReportIssue(Diagnostic.Create(Rule, method.Identifier.GetLocation()));
+                        }
+                    }
+                },
+                SyntaxKind.MethodDeclaration);
+
+        private class Walker : SafeCSharpSyntaxWalker
+        {
+            public bool HasInvocationOutsideTryCatch { get; private set; }
+
+            public override void Visit(SyntaxNode node)
+            {
+                if (!HasInvocationOutsideTryCatch)  // Stop walking when we know the answer
+                {
+                    switch (node.Kind())
+                    {
+                        case SyntaxKind.AnonymousMethodExpression:
+                        case SyntaxKind.SimpleLambdaExpression:
+                        case SyntaxKind.ParenthesizedLambdaExpression:
+                        case SyntaxKindEx.LocalFunctionStatement:
+                            return;
+                    }
+                    base.Visit(node);
+                }
+            }
+
+            public override void VisitInvocationExpression(InvocationExpressionSyntax node) =>
+                HasInvocationOutsideTryCatch = true;
+
+            public override void VisitTryStatement(TryStatementSyntax node)
+            {
+                if (!node.Catches.AsEnumerable().Any(CatchesAllExceptions))
```

Comment:
```suggestion
                if (!node.Catches.Any(CatchesAllExceptions))
```

Reply:
Hmm, I wonder why it didn't work when I tried for the first time. There was a clash with `SyntaxList.Any()` method